pkg/operator/ceph/spec/spec.go

```diff
@@ -42,10 +42,20 @@ var logger = capnslog.NewPackageLogger("github.com/rook/rook", "ceph-spec")
 
 // return the volume and matching volume mount for mounting the config override ConfigMap into
 // containers as "/rook/ceph/ceph.conf".
-func configOverrideConfigMapVolumeAndMount() (v1.Volume, v1.VolumeMount) {
+func configOverrideConfigMapVolumeAndMount(externalCluster bool) (v1.Volume, v1.VolumeMount) {
 	name := k8sutil.ConfigOverrideName // configmap name and name of volume
 	dir := config.EtcCephDir
 	file := "ceph.conf"
+	var isOptional bool
+
+	// The configmap "rook-config-override" must be false when running an external cluster configuration
+	// This is needed so that the external cluster can bootstrap resources in its namespace
+	// If the configmap is required, the pods will wait forever for it to exist
+	// On an external setup, they are no OSDs, so there is no way this cm will exist
+	if externalCluster {
+		isOptional = true
```

Comment:
Wait, why do we need to make this CM optional? Can't we create the CM for external clusters too? The user should still be able to add custom ceph settings there.

Reply:
So you're suggesting, creating an empty cm instead of making it optional? That makes sense too, I was actually hesitating at first.